### PR TITLE
Feat : Added Homepage

### PR DESCRIPTION
**Pull request by srejitk**
This Feature request adds the following features:

1. Auth
2. Home Page
3. Sign In
4. Sign Up
5. Toast
6. Editor Component

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export default function Header() {
+  const [showOption, setShowOption] = useState(false);
+  const { isLogged, logoutHandler } = useAuth();
+
+  const optionHandler = () => {
+    setShowOption(!showOption);
+  };
+
+  console.log(showOption);
```
Comment (kjais1720) [inline src/components/Header/Header.jsx]: remove the log

**Review by therajatg — APPROVED** [review]
code looks good

File: src/pages/Signup/Signup.jsx [hunk 2]
```diff
@@ -1,5 +1,166 @@
...
+        console.log("HOAGAYA");
+        localStorage.setItem("Token", response.data.encodedToken);
+      }
+    } catch (error) {
+      setError("Something went wrong");
+      console.log(error);
+    }
+  };
+  const pwdVisibiltyHandler = () => {
+    setShowPassword(!showPassword);
```
Comment (kjais1720) [inline src/pages/Signup/Signup.jsx]: here, do a functional update instead

**Review by kjais1720 — APPROVED** [review]
The code looks good overall, just a couple of minor changes suggested

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (srejitk) [reply src/components/Header/Header.jsx]: Thanks for pointing this out. Have removed it.

File: src/pages/Signup/Signup.jsx (hunk 2 above)
Comment (srejitk) [reply src/pages/Signup/Signup.jsx]: Thank You , have corrected it in the latest commit

**Pull request merged**